### ISSUE TITLE
Stacktrace for telnet_encrypt_overflow

### ISSUE THREAD

**Issue by leebaird** [rapid7/metasploit-framework#8060]
## Steps to reproduce

```
resource (external/master.rc)> use auxiliary/scanner/telnet/telnet_encrypt_overflow
resource (external/master.rc)> run
[-] 10.0.0.249:23      - 10.0.0.249:23 Error: Connection reset by peer ["<internal:prelude>:76:in `__read_nonblock'", "<internal:prelude>:76:in `read_nonblock'", "/usr/share/metasploit-framework/vendor/bundle/ruby/2.3.0/gems/rex-core-0.1.7/lib/rex/io/stream.rb:72:in `read'", "/usr/share/metasploit-framework/vendor/bundle/ruby/2.3.0/gems/rex-core-0.1.7/lib/rex/io/stream.rb:202:in `get_once'", "/usr/share/metasploit-framework/lib/msf/core/exploit/telnet.rb:161:in `recv_telnet'", "/usr/share/metasploit-framework/lib/msf/core/exploit/telnet.rb:147:in `recv'", "/usr/share/metasploit-framework/lib/msf/core/exploit/telnet.rb:126:in `block in connect'", "/usr/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'", "/usr/lib/ruby/2.3.0/timeout.rb:106:in `timeout'", "/usr/share/metasploit-framework/lib/msf/core/exploit/telnet.rb:124:in `connect'", "/usr/share/metasploit-framework/modules/auxiliary/scanner/telnet/telnet_encrypt_overflow.rb:43:in `block in run_host'", "/usr/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'", "/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'", "/usr/lib/ruby/2.3.0/timeout.rb:106:in `timeout'", "/usr/share/metasploit-framework/modules/auxiliary/scanner/telnet/telnet_encrypt_overflow.rb:42:in `run_host'", "/usr/share/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:135:in `block (2 levels) in run'", "/usr/share/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'"]
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

### Metasploit version
v4.14.0-dev

### I installed Metasploit with:
Kali package via apt

What OS are you running Metasploit on?
Kali 2016.2 rolling

**Comment by adv16** [comment]
this looks similar to #8061  @leebaird  are you able to reproduce these?

**Comment by busterb** [comment]
Follow the breadcrumbs on #8061, this should be the same fix, agree with @adv16

**Comment by itsmeroy2012** [comment]
I'm unable to reproduce this issue @leebaird . This will have the same fix as in the referenced issue #8061 right? @busterb 

I modified the telnet_encrypt_overflow file:-
```
rescue ::Rex::ConnectionError, ::Errno::ECONNRESET => e
      print_error("A network issue has occurred: #{e.message}")
      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}") 
rescue Timeout::Error
      print_error("#{target_host}:#{rport} Timed out after #{to} seconds")
      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
rescue ::Exception => e
      print_error("#{target_host}:#{rport} Error: #{e} #{e.backtrace}")
      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```

## Steps to produce:
```
msf > use auxiliary/scanner/telnet/telnet_version
msf auxiliary(telnet_version) > use auxiliary/scanner/telnet/telnet_encrypt_overflow
msf auxiliary(telnet_encrypt_overflow) > set RHOSTS 127.0.0.1
RHOSTS => 127.0.0.1
msf auxiliary(telnet_encrypt_overflow) > set RPORT 21
RPORT => 21
msf auxiliary(telnet_encrypt_overflow) > exploit

[-] 127.0.0.1:21       - A network issue has occurred: The connection was refused by the remote host (127.0.0.1:21).
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```
Although I am not able to reproduce the issue but is this the proper fix? I'll put up a PR if acknowledged.
